src/rules/custom-property-no-empty-line-before/README.md

```diff
@@ -0,0 +1,83 @@
+# custom-property-no-empty-lines-before
+
+Disallow empty lines before custom properties.
+
+```css
+a { 
+  --foo-bar: 1px;
+                      /* ← */
+  --foo-baz: 2px;     /* ↑ */   
+}                     /* ↑ */
+/**                      ↑
+  *              This line */
+```
+
+## Options
+
+### `true`
+
+The following patterns are considered warnings:
+
+```css
+a { 
+  
+  --foo-bar: 1px;
+  --foo-baz: 2px; 
+}
+```
+
+```css
+a {
+  --foo-bar: 1px;
+
+  --foo-baz: 2px; 
+}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a { 
+  --foo-bar: 1px;
+  --foo-baz: 2px; 
+}
+```
+## Optional options
+
+### `expect: ["first-after-non-custom-property-sibling"]`
+
+Require an empty line for first custom property after a non custom property.
```

Comment:
> (Is that right? Like, after nested rules and at-rules? What about comments?)

Good point. We've been specific up until this rule e.g. `"after-comment"` etc. I don't think being broad-sweeping in our categorisation is going to work as comments complicate things.

Shall we be specific and just replace this option with `except: ["after-comment"]` for now? As custom properties generally appear at the top of the declaration block I doubt they'll be much call for any other `except` options. However, we can add specific ones as they are requested e.g. `"after-at-rule"`:

```css
a {
  @extends b;
  --custom-prop: value;
}
```